echopype/commongrid/mvbs.py

```diff
@@ -417,45 +441,34 @@ def get_MVBS_along_channels(
         A Dataset containing ``Sv`` and ``echo_range`` data with coordinates
         ``channel``, ``ping_time``, and ``range_sample``
     echo_range_interval: np.ndarray
-        1D array (used by np.digitize) representing the binning required for ``echo_range``
+        1D array (used by np.digitize) representing
+        the binning required for ``echo_range``
     ping_interval: np.ndarray
-        1D array (used by np.digitize) representing the binning required for ``ping_time``
+        1D array (used by np.digitize) representing
+        the binning required for ``ping_time``
 
     Returns
     -------
-    np.ndarray
-        The MVBS value of the input ``ds_Sv`` for all channels
-
-    Notes
-    -----
-    If the values in ``ds_Sv`` are delayed then the binning and mean of ``Sv`` with
-    respect to ``echo_range`` will take place, then the delayed result will be computed,
-    and lastly the binning and mean with respect to ``ping_time`` will be completed. It
-    is necessary to apply a compute midway through this method because Dask graph layers
-    get too large and this makes downstream operations very inefficient.
+    xr.Dataset
+        The MVBS dataset of the input ``ds_Sv`` for all channels
     """
 
-    all_MVBS = []
-    for chan in ds_Sv.channel:
-        # squeeze to remove "channel" dim if present
-        # TODO: not sure why not already removed for the AZFP case. Investigate.
-        ds = ds_Sv.sel(channel=chan).squeeze()
-
-        # average should be done in linear domain
-        sv = 10 ** (ds["Sv"] / 10)
-
-        # get MVBS for channel in linear domain
-        chan_MVBS = bin_and_mean_2d(
-            sv.data,
-            bins_time=ping_interval,
-            bins_er=echo_range_interval,
-            times=sv.ping_time.data,
-            echo_range=ds["echo_range"],
-            comprehensive_er_check=True,
-        )
-
-        # apply inverse mapping to get back to the original domain and store values
-        all_MVBS.append(10 * np.log10(chan_MVBS))
+    # average should be done in linear domain
+    sv = ds_Sv["Sv"].pipe(_linear_transform)
+
+    # reduce along ping_time and echo_range
+    # by binning and averaging
+    mvbs = xarray_reduce(
+        sv,
+        sv["channel"],
+        ds_Sv["ping_time"],
+        ds_Sv["echo_range"],
+        func="mean",
+        expected_groups=(None, ping_interval, echo_range_interval),
+        isbin=[False, True, True],
+        method="map-reduce",
```

Comment:
Maybe this should be exposed to the user..? After some conversation with @leewujung, "blockwise" might be better.